syntax = "proto3";

option go_package = "capstone-2025-30/proto";

import "error.proto";

message InquiryRequest{
  string store_code = 1; // 가게 code
  repeated float frame_data = 2; // 영상 한 프레임 좌표 데이터
  string inquiry_type = 3; // 문의 종류(주문 문의사항 / 일반 문의사항)
  int32 num = 4; // 주문 번호 or 일반 문의 번호
}

message InquiryResponse{
  bool success = 1; // 성공 여부
  optional EError error = 2; // 에러 코드
}

message FastInquiryRespIsNoRequest{
  string store_code = 1; // 가게 코드
  string title = 2; // 주문 대화창인가(order) / 일반 문의사항 대화창인가 (inquiry)
  int32 num = 3;
}

message FastInquiryRespIsNoResponse{
  bool success = 1; // 성공 여부
  optional EError error = 2; // 에러 코드
}